polkadot/xcm/xcm-simulator/fuzzer/src/fuzz.rs

```diff
@@ -148,13 +151,39 @@ pub fn relay_ext() -> sp_io::TestExternalities {
 pub type RelayChainPalletXcm = pallet_xcm::Pallet<relay_chain::Runtime>;
 pub type ParachainPalletXcm = pallet_xcm::Pallet<parachain::Runtime>;
 
+// We check XCM messages recursively for blocklisted messages
+fn recursively_matches_blocklisted_messages(message: &mut Instruction<()>) -> bool {
+	match message {
+		DepositReserveAsset { xcm, .. } |
+		ExportMessage { xcm, .. } |
+		InitiateReserveWithdraw { xcm, .. } |
+		InitiateTeleport { xcm, .. } |
+		TransferReserveAsset { xcm, .. } |
+		SetErrorHandler(xcm) |
+		SetAppendix(xcm) =>
+			Vec::from(xcm.inner()).iter_mut().any(recursively_matches_blocklisted_messages),
```

Comment:
You could still use iterators to avoid any clones, but its fine 🙏

Reply:
Good point, addressed in 11302c9a139ac04c687dfce8cb5c4104d67f4bec